Docs: added architecture overview

shahednasser opened the pull request:
Added architecture overview on the introduction page

olivermrbl reviewed (commented):
Really nice @shahednasser ! Would it make sense to only have the title and diagram on the homepage and then move the detailed description into it's own section in the sidebar? 

My reasoning is, that I would like to keep the guides, tutorials, setup above the fold when you enter our documentation.

<img width="1095" alt="image" src="https://user-images.githubusercontent.com/59018053/155487638-478559fa-9b87-462b-88e3-181f5851c353.png">

Wdyt?

shahednasser commented:
Ok so since there are a couple of sections I want to add to the introduction, I changed the current introduction to "Overview" and I added a new introduction page with the Architecture Overview in details. Let me know what you think!

olivermrbl reviewed (approved):
LGTM!
